code/src/reducers/tasks.js

```diff
@@ -0,0 +1,50 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+	list: {
+		name: 'To Do List',
+		items: [
+			// {
+			// 	id: 0,
+			// 	text: 'Eat breakfast',
+			// 	isComplete: true,
+			// },
+			// {
+			// 	id: 1,
+			// 	text: 'Go to gym',
+			// 	isComplete: false,
+			// },
+			// {
+			// 	id: 3,
+			// 	text: 'Take a nap',
+			// 	isComplete: false,
+			// },
+		],
+	},
+};
+
+const tasks = createSlice({
+	name: 'tasks',
+	initialState: initialState,
+	reducers: {
+		addTask: (state, action) => {
+			const itemInfo = action.payload;
+			state.list.items.push(itemInfo);
+		},
+		toggleDone: (state, action) => {
+			const { itemIndex, isComplete } = action.payload;
+			state.list.items[itemIndex].isComplete = isComplete;
+		},
+		deleteTask: (state, action) => {
+			const { itemIndex } = action.payload;
+			state.list.items = state.list.items.filter(
+				(item, index) => index !== itemIndex
+			);
+		},
+		clearAll: (state) => {
```

Comment:
is this functionality implemented? I couldn't find it anywhere, but a nice feature if was implemented!